python/plugins/processing/tools/postgis.py

```diff
@@ -138,41 +163,54 @@ def _quote(self, ident):
 
 class GeoDB:
 
+    @classmethod
+    def from_name(cls, name):
+        uri = uri_from_name(name)
+        return cls(uri=uri)
+
     def __init__(self, host=None, port=None, dbname=None, user=None,
-                 passwd=None):
+                 passwd=None, service=None, uri=None):
         # Regular expression for identifiers without need to quote them
         self.re_ident_ok = re.compile(r"^\w+$")
 
-        self.host = host
-        self.port = port
-        self.dbname = dbname
-        self.user = user
-        self.passwd = passwd
-
-        if self.dbname == '' or self.dbname is None:
-            self.dbname = self.user
-
-        try:
-            self.con = psycopg2.connect(self.con_info())
-        except psycopg2.OperationalError as e:
-            raise DbError(unicode(e))
+        if uri:
+            self.uri = uri
+        else:
+            self.uri = QgsDataSourceURI()
+            if service:
+                self.uri.setConnection(service, dbname, user, passwd)
+            else:
+                self.uri.setConnection(host, port, dbname, user, passwd)
+
+        conninfo = self.uri.connectionInfo(False)
+        for i in range(3):
+            try:
+                self.con = psycopg2.connect(self.uri.connectionInfo(True))
```

Comment:
sorry to be late to add this comments... btw this support can be added as bug fix. Just to say that my comments are not blocking the PR.